dss_metrics/minio_rest_collector.py

```diff
@@ -0,0 +1,132 @@
+        self.filter = filter
+        self.minio_metrics = {'minio_disk_storage_used_bytes'}
+        self.metrics_url_prefix = "http://"
+        self.metrics_url_suffix = "/minio/prometheus/metrics"
+        self.mc = configs['mc_binary_path']
+        self.conf_json_bucket_suffix = configs['conf_json_bucket_suffix']
+        self.cluster_id = self.configs['cluster_id']
+        self.TYPE = 'minio'
+
+    def poll_statistics(self, metrics_data_buffer):
+        cluster_endpoint_map = self.get_miniocluster_endpoint_map()
+        for id, endpts in cluster_endpoint_map.items():
+            miniocluster_id = id
+            minio_endpoint = endpts.pop()
+            minio_metrics = self.get_minio_metrics_from_endpoint(
+                minio_endpoint)
+
+            tags = {}
+            tags['cluster_id'] = self.cluster_id
+            tags['target_id'] = socket.gethostname()
+            tags['minio_id'] = miniocluster_id
+            tags['type'] = self.TYPE
+
+            for metric in minio_metrics:
+                if self.filter and not self.check_whitelist_key(metric[0]):
+                    continue
+                metrics_data_buffer.append(
+                    metrics.MetricInfo(
+                        metric[0], metric[0], metric[1], tags, time.time())
+                )
+
+    def check_whitelist_key(self, key):
+        for regex in self.whitelist_patterns:
+            if re.match(regex, key):
+                return True
+        return False
+
+    def get_miniocluster_endpoint_map(self):
+        get_hosts_cmd = self.mc + " " + "config host list"
+        proc = subprocess.Popen(get_hosts_cmd.split(' '),
+                                stdout=subprocess.PIPE)
+        local_minio_host = None
+        miniocluster_endpoint_map = {}  # {<miniocluster> : {<endpoints>} }
+
+        for line in proc.stdout.readlines():
+            # find local minio host or endpoint
+            decoded_line = line.decode('utf-8').strip("\n")
+            if decoded_line.startswith('local_'):
+                local_minio_host = decoded_line
+                break
+
+        if local_minio_host:
+            conf_json_path = local_minio_host + self.conf_json_bucket_suffix
+            get_conf_json_cmd = self.mc + " cat " + conf_json_path
+            proc = subprocess.Popen(get_conf_json_cmd.split(' '),
+                                    stdout=subprocess.PIPE)
+            dss_conf_dict = json.loads(proc.communicate()[0].decode('utf-8'))
+
+            try:
+                for cluster in dss_conf_dict["clusters"]:
+                    minio_cluster_id = cluster["id"]
+                    minio_endpoints = set()
+                    for endpoint_info in cluster["endpoints"]:
+                        minio_endpoints.add(
+                            endpoint_info["ipv4"] + ":"
+                            + str(endpoint_info["port"])
+                        )
+                    miniocluster_endpoint_map[minio_cluster_id] = (
```

Comment:
no brackets are required for minio_endpoints. redundant

Reply:
parenthesis needed to have var on next line due to Pep8 line length requirement.

Reply:
why do you need to send that single var to the next line. you can do something like
` miniocluster_endpoint_map[minio_cluster_id] = minio_endpoints`

Reply:
@grandsuri it is because the line gets too long if I do it that way and it exceeds 80 characters